Sync v2-5-stable with v2-5-test to release 2.5.0

ephraimbuddy opened the pull request:
Time for `2.5.0rc1`!

ephraimbuddy committed:
Update version to 2.5.0

ephraimbuddy committed:
Add release notes

raphaelauv commented on README.md:
```diff
@@ -86,7 +86,7 @@ Airflow is not a streaming solution, but it is often used to process real-time d
 Apache Airflow is tested with:
 
-|                     | Main version (dev)           | Stable version (2.4.2)       |
+|                     | Main version (dev)           | Stable version (2.5.0)       |
```
missing PostgreSQL 15 support for 2.5.0

potiuk replied:
good catch

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,279 @@
 
 .. towncrier release notes start
 
+Airflow 2.5.0 (2022-11-28)
```
```suggestion
Airflow 2.5.0 (2022-11-30)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,279 @@
+Significant Changes
+^^^^^^^^^^^^^^^^^^^
+
+- ``airflow dags test`` no longer performs a backfill job.
```
```suggestion
``airflow dags test`` no longer performs a backfill job (#26400)
""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,279 @@
+
+  In order to make ``airflow dags test`` more useful as a testing and debugging tool, we no
+  longer run a backfill job and instead run a "local task runner". Users can still backfill
+  their DAGs using the ``airflow dags backfill`` command. (#26400)
```
```suggestion
  their DAGs using the ``airflow dags backfill`` command.
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,279 @@
+  In order to make ``airflow dags test`` more useful as a testing and debugging tool, we no
+  longer run a backfill job and instead run a "local task runner". Users can still backfill
+  their DAGs using the ``airflow dags backfill`` command. (#26400)
+- Airflow config section ``kubernetes`` renamed to ``kubernetes_executor``
```
```suggestion
Airflow config section ``kubernetes`` renamed to ``kubernetes_executor`` (#26873)
"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,279 @@
+  their DAGs using the ``airflow dags backfill`` command. (#26400)
+- Airflow config section ``kubernetes`` renamed to ``kubernetes_executor``
+
+  KubernetesPodOperator no longer considers any core kubernetes config params, so this section now only applies to kubernetes executor. Renaming it reduces potential for confusion. (#26873)
```
```suggestion
  KubernetesPodOperator no longer considers any core kubernetes config params, so this section now only applies to kubernetes executor. Renaming it reduces potential for confusion.
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,279 @@
+  KubernetesPodOperator no longer considers any core kubernetes config params, so this section now only applies to kubernetes executor. Renaming it reduces potential for confusion. (#26873)
+- ``ExternalTaskSensor`` no longer hangs indefinitely when ``failed_states`` is set, an ``execute_date_fn`` is used, and some but not all of the dependent tasks fail. Instead, an ``AirflowException`` is thrown as soon as any of the dependent tasks fail.
+
+  Any code handling this failure in addition to timeouts should move to cathing the ``AirflowException`` baseclass and not only the ``AirflowSensorTimeout`` subclass. (#27190)
```
This will need a similar change as above, but we probably need a shorter title for this?

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,279 @@
+- docs for custom view using plugin (#27244)
+- Update SLA wording to reflect it is relative to Dag Run start. (#27111)
+- Add missed import in the Trigger Rules example (#27309)
+- Update dags.rst (#27110)
```
Typo, can skip this one.
```suggestion
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,279 @@
+- Update SLA wording to reflect it is relative to Dag Run start. (#27111)
+- Add missed import in the Trigger Rules example (#27309)
+- Update dags.rst (#27110)
+- Fix typo in ``set-up-database.rst`` (#27137)
```
```suggestion
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,279 @@
+- Don't re-patch pods that are already controlled by current worker (#26778)
+- Add separate error handler for 405(Method not allowed) errors (#26880)
+- Avoid 500 on dag redirect (#27064)
+- Make tracebacks opt-in (#27059)
```
```suggestion
```

Hmm, these were already in 2.4.2? Might be others as well.

ephraimbuddy replied:
Yeah. The changelog command was getting everything since 2.4.1. I have manually sorted the bugfixes

mobuchowski committed:
tests: always cleanup registered test listeners (#27896)

Signed-off-by: Maciej Obuchowski <obuchowski.maciej@gmail.com>

Signed-off-by: Maciej Obuchowski <obuchowski.maciej@gmail.com>
(cherry picked from commit eba04d7c400c0d89492d75a7c81d21073933cd0c)

uranusjr committed:
Remove is_mapped attribute (#27881)

(cherry picked from commit 3e288abd0bc3e5788dcd7f6d9f6bef26ec4c7281)

blag committed:
Soft delete datasets that are no longer referenced in DAG schedules or task outlets (#27828)

* Soft delete datasets that are no longer referenced anywhere

* Move the orphaning logic into the scheduler and adjust config option name accordingly

* Rename config option scheduler.cleanup_interval -> scheduler.parsing_cleanup_interval

* Include default column value in migration

* deprecate old interval config; move code to scheduler_job

* First pass at a test

* Fix migration

* Apply suggestions from code review

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>

* Don't batch migrations if we don't need to

* Revert "Don't batch migrations if we don't need to" - gotta batch migrations for SQLite

This reverts commit 652f7452d3b418c991d409a2b0fc041443048545.

* Tweak migrations

* Use sqlalchemy.sql.True_() to support all DB backends

* Various cleanups

* Add test for un-orphaning datasets once they are referenced again

* Use sqlalchemy.sql.expression.true()

Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

* Fix orphaning datasets on MSSQL

* Comment the un-orphan process and use sqla.sql.expression.false()

* Add newsfragment about renamed config option

* add mssql_drop_default flag

* Use server_default in the ORM as well

* Defensively clear datasets before and after DAG tests

* Reconcile migration with ORM model

* Remove now erroneous comment

* Change to use server_default='0'

* Update airflow/configuration.py

Co-authored-by: Jed Cunningham <jedcunningham@apache.org>
Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>
Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>
(cherry picked from commit 3fef6a47834b89b99523db6d97d6aa530657a008)

ephraimbuddy committed:
Mask audit log extra field (#27923)

(cherry picked from commit 1e73b1cea2d507d6d09f5eac6a16b649f8b52522)

bolkedebruin committed:
Add allow list for imports during deserialization (#27887)

During deserialization Airflow can instantiate arbitrary
objects for which it imports modules. This can be dangerous
as it could lead to unwanted effects. With this change
administrators can now limit what objects can be deserialized.
It defaults to Airflow's own only.

(cherry picked from commit 542cfdc270aeb9b05ad150df28bc48b0a84c0f38)

ashb committed:
Include correct meta tag to get dataset events (#27927)

Without this the section on "Dataset updates caused by this task
instance" would show no events!

(cherry picked from commit 8588d134746045cc1d2c3621fb34e18c883cbf67)
